game-core/src/main/java/games/strategy/triplea/delegate/Matches.java

```diff
@@ -2033,14 +2026,13 @@ private Matches() {}
   }
 
   static Predicate<Unit> unitHasWhenCombatDamagedEffect(final String filterForEffect) {
-    return u -> {
-      if (!unitHasWhenCombatDamagedEffect().test(u)) {
+    return unit -> {
+      if (!unitHasWhenCombatDamagedEffect().test(unit)) {
         return false;
       }
```

Comment:
```suggestion
    return unitHasWhenCombatDamagedEffect().and(unit -> {
```

For style points

Reply:
A number of these I noticed. Tried to keep the scope as limited as possible to avoid getting too deep and mixing too many different kinds of changes in.

Reply:
This update resulted in:
```
Step 'google-java-format' found problem in 'game-core/src/main/java/games/strategy/triplea/delegate/Matches.java':
```

The inline suggestion is nice, though I think we probably should reserve it for quick typos. This PR is really risky for scope creep, there are tons of follow on simplifications that should be done.

Reply:
Needed a missing close paren.

I like the change, but it's not for this PR. I don't like how the indentation is updated and increases the diff pretty drastically. Collapsing methods from Matches into TripleAPlayer was arguably too far of an update as well. With that the focus is really just on the one change at a time.